libnetwork/drivers/bridge/port_mapping_linux.go

```diff
@@ -134,6 +146,33 @@ func (n *bridgeNetwork) addPortMappings(
 		}
 	}
 
+	// Now the iptables rules are set up, it's safe to start the userland proxy.
+	// (If it was started before the iptables rules were created, it may have
+	// accepted a connection, then become unreachable due to NAT rules sending
+	// packets directly to the container.)
+	somaxconn := getSOMAXCONN()
+	for i := range bindings {
+		if bindings[i].boundSocket == nil {
+			continue
+		}
+		if bindings[i].Proto == types.TCP {
+			// If not starting the proxy, nothing will ever accept a connection on the
+			// socket. But, listen anyway so that the binding shows up in "netstat -at".
+			if err := syscall.Listen(int(bindings[i].boundSocket.Fd()), somaxconn); err != nil {
```

Comment:
```suggestion
			if err := syscall.Listen(int(bindings[i].boundSocket.Fd()), 0); err != nil {
```
Why tell the kernel to queue up the maximum number of connections possible, consuming kernel resources, when we have no intention to ever accept() any of them? [Zero is a valid backlog value.](https://github.com/torvalds/linux/commit/64a146513f8f)

Reply:
That's what `dummyProxy` did but, right, there's no need.